### PR TITLE
 Silence hints/recommendations from Docker in the console 

### LINKED ISSUE #1635: Silence the recommendations/hints from Docker Desktop
### Describe the bug

Using Docker, as of late, includes a lot of unhelpful recommendations being printed to the console.

For instance:
```
(venv) user@vm-win11:~/tmp/beeware/helloworld$ briefcase build --target fedora

[Docker] Downloading Docker base image for fedora...
Using default tag: latest
latest: Pulling from library/fedora
8237fce9fd6b: Pull complete
Digest: sha256:06df381d697d14940c886fda8e94a4fdc838df74e93f65111ed3ea04f7a7d6e0
Status: Downloaded newer image for fedora:latest
docker.io/library/fedora:latest

What's Next?
  View a summary of image vulnerabilities and recommendations → docker scout quickview fedora
```

We should disable these given they aren't useful for our use-case.

### Steps to reproduce

Run Docker-related commands with Docker Desktop.

### Expected behavior

Use environment variable `DOCKER_CLI_HINTS=false` to disable this output.

### Screenshots

_No response_

### Environment

- Operating System: Windows 11
- Python version: 3.8
- Software versions:
  - Briefcase: `0.3.17.dev283+g23daa64e`

### Logs

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rmartin16**
## Changes
- Resolves #1635

## Notes
- Dependent on #1603

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [X] I have read the **CONTRIBUTING.md** file
- [X] I will abide by the code of conduct

### CODE REVIEW

**Comment by rmartin16** [comment]
This approach adds a `final_kwargs()` to `Docker` and assumes it will be used anytime `subprocess` is being used to invoke `docker`. This excludes the `docker` invocations during `verify()`...but they could be added if `final_kwargs()` is made a Class method. I avoided doing this, though, because I'd also like to create a `Protocol` for all these interfaces that are modeled on `Subprocess`.

At any rate, another approach I considered was updating `dockerize_args()` to support invoking `docker` for commands other than `docker run`. While that works, it kinda started to overload `dockerize_args()`, IMO.

Thoughts or other ideas welcome.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
The general approach looks good. My only real hesitation is around the name "final" and whether it's maybe one level of abstraction too many to be worth pulling that out. Yes, it's a repeated configuration; but given that it's only being used in 2 places, that isn't likely to increase (AFAICT), and being an instance method means it can't be easily used for verification... maybe it's just easier to include the extra environment variable in the 2 places it's needed.

**Comment by rmartin16** [comment]
I agree `final_kwargs()` may be misleading since this is all still wrapping `subprocess`. So, what about just an encapsulation for the `env` for Docker calls to `subprocess`? This way, it'll be trivial to add additional env vars in the future and it can be added to even the verification `docker` calls.

File: src/briefcase/integrations/docker.py [hunk 1]
```diff
@@ -388,6 +399,18 @@ def check_output(self, args: SubprocessArgsT, image_tag: str, **kwargs) -> str:
...
+    @classmethod
+    def subprocess_env(cls, env: dict[str, str] | None = None) -> dict[str, str]:
+        """Adds environment variables to the context that Docker runs in."""
+        env = {} if env is None else env
+        env.update(
+            {
+                # Disable the hints/recommendations that Docker prints in the console
+                "DOCKER_CLI_HINTS": "false",
+            }
+        )
```
Comment (freakboy3742) [inline src/briefcase/integrations/docker.py:411]: Two minor nits here: 

1. If the user provides an `env` dictionary as an argument, that dictionary will be modified in place - so using an environment in a subprocess will alter the environment
2. If, for some reason, you wanted to *enable* Docker hints (I don't know why, but work with me here 😄 ), there's no way to do so.

What about reversing the direction here: starting with a dictionary that contains `DOCKER_CLI_HINTS: "False"``, and updating *that* dictionary with env. That will ensure that the final dictionary is a copy, and any explicitly user-provided value will override the default.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
This approach looks much better to me. One minor nit inline.

File: src/briefcase/integrations/docker.py (hunk 1 above)
Comment (rmartin16) [reply src/briefcase/integrations/docker.py]: I should have realized this wouldve updated the dictionary input...which is definitely undesired. I updated it so this (and future) default can be overridden.

**Review by freakboy3742 — APPROVED** [review]
👍

**Pull request merged**